lib/runtime/wasmer/exports.go

```diff
@@ -86,6 +86,60 @@ func (in *Instance) GrandpaAuthorities() ([]types.Authority, error) {
 	return types.GrandpaAuthoritiesRawToAuthorities(gar)
 }
 
+// BabeGenerateKeyOwnershipProof returns babe key ownership proof from runtime.
+func (in *Instance) BabeGenerateKeyOwnershipProof(slot uint64, authorityID [32]byte) (
+	types.OpaqueKeyOwnershipProof, error) {
+
+	combinedArg := []byte{}
+	encodedSetID, err := scale.Marshal(slot)
+	if err != nil {
+		return nil, fmt.Errorf("encoding set id: %w", err)
+	}
+	combinedArg = append(combinedArg, encodedSetID...)
+
+	encodedAuthorityID, err := scale.Marshal(authorityID)
+	if err != nil {
+		return nil, fmt.Errorf("encoding authority id: %w", err)
+	}
+	combinedArg = append(combinedArg, encodedAuthorityID...)
+
+	ret, err := in.Exec(runtime.BabeAPIGenerateKeyOwnershipProof, combinedArg)
+	if err != nil {
+		return nil, err
+	}
+
+	keyOwnershipProof := types.OpaqueKeyOwnershipProof{}
+	err = scale.Unmarshal(ret, &keyOwnershipProof)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshalling: %w", err)
+	}
+
+	return keyOwnershipProof, err
+}
+
+// BabeSubmitReportEquivocationUnsignedExtrinsic reports equivocation report to the runtime.
+func (in *Instance) BabeSubmitReportEquivocationUnsignedExtrinsic(
+	equivocationProof types.BabeEquivocationProof, keyOwnershipProof types.OpaqueKeyOwnershipProof,
+) error {
+
+	combinedArg := []byte{}
+
+	encodedEquivocationProof, err := scale.Marshal(equivocationProof)
+	if err != nil {
+		return fmt.Errorf("encoding equivocation proof: %w", err)
+	}
+	combinedArg = append(combinedArg, encodedEquivocationProof...)
+
+	encodedKeyOwnershipProof, err := scale.Marshal(keyOwnershipProof)
+	if err != nil {
+		return fmt.Errorf("encoding key ownership proof: %w", err)
+	}
+	combinedArg = append(combinedArg, encodedKeyOwnershipProof...)
+
+	_, err = in.Exec(runtime.BabeAPISubmitReportEquivocationUnsignedExtrinsic, combinedArg)
+	return err
```

Comment:
```suggestion
	if err != nil {
		return fmt.Errorf("execing %s: %w", runtime.BabeAPISubmitReportEquivocationUnsignedExtrinsic, err)
	}
	return nil
```

Reply:
Instead of wrapping the errors here, why not have the caller just wrap it? Feels a little redundant to me, but also a genuine question. My intuition is that the returned error will be enough for the caller (or us reading logs) to know what is wrong. For example, a formatting error is an error from the runtime which is logged, so we don't need that info in the log because its already obvious. 

But like I said, genuine question! Just wanna make sure we are adding things for good reason

Reply:
> Instead of wrapping the errors here, why not have the caller just wrap it?

Yep, that goes with the idea of adding context to errors while returning errors. So, function names should already be there in the returned error.

There isn't any need for changes here, because of that.